pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -671,9 +671,58 @@ func getServerByName(client *gophercloud.ServiceClient, name types.NodeName) (*S
 	return &serverList[0], nil
 }
 
-func nodeAddresses(srv *servers.Server, networkingOpts NetworkingOpts) ([]v1.NodeAddress, error) {
+// IP addresses order:
+// * interfaces private IPs
+// * access IPs
+// * metadata hostname
+// * server object Addresses
```

Comment:
This should be `server object Addresses(floating type)` to be more specific

Reply:
added